### PR TITLE
Automate and document releasing to Hackage + couple cleanups

### PR DESCRIPTION
**Pull request by liskin**
#### [Revise dependencies a bit](../commit/96baa79c3548af02c95f90a1b907e50740c8680a)

utf8-string isn't needed since d9e3ebf53194d1cf02bae28f0c885b5c75610cfa (late 2014).

Most of the executable xmonad deps are unnecessary since 307b82a53d519f5c86c009eb1a54044a616e4a5c (early 2015).

Switch from data-default to data-default-class to avoid bringing in deps and instances we don't need.

#### [Revise extra-source-files in xmonad.cabal](../commit/061b1bde977233a5b7ad2656c900f0d981885733)

Ship the new docs, drop tests as these get picked up automatically.

#### [Automate and document releasing to Hackage](../commit/9640ca7077e771d54272f6e0ce923085c81d3c77)

Pioneered in https://github.com/xmonad/X11, this adds automation for Hackage releases and updates MAINTAINERS.md with simplified instruction for the release procedure.

Related: https://github.com/xmonad/xmonad/issues/75
Related: https://github.com/xmonad/xmonad/issues/264
Related: https://github.com/xmonad/xmonad-contrib/issues/393

### Checklist

  - [X] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [X] I've confirmed these changes don't belong in xmonad-contrib instead

  - n/a I updated the `CHANGES.md` file

### CODE REVIEW

File: MAINTAINERS.md [hunk 1]
```diff
@@ -59,58 +59,58 @@
...
+  2. Review documentation files and make sure they are accurate:
 
-  2. Update the version number in the `*.cabal` files and verify
-     dependencies and documentation.  This includes the `tested-with:`
-     field.
+     - [`README.md`](README.md)
+     - [`CHANGES.md`](CHANGES.md)
+     - [`INSTALL.md`](INSTALL.md)
+     - [`man/xmonad.1.markdown.in`](man/xmonad.1.markdown.in)
+     - haddocks <!-- TODO: link to web with doc snapshots once there's one -->
```
Comment (slotThe) [inline MAINTAINERS.md]: There is one now! :)

File: MAINTAINERS.md [hunk 2]
```diff
@@ -59,58 +59,58 @@
...
-  9. Merge the release branches into `master`
+     - Post a [new release announcement][web-announce]
+     - Check install instructions, guided tour, keybindings cheat sheet, …
 
-  10. Update the website:
+  7. Send e-mail announcement to:
 
-      * Generate and push haddocks with `xmonad-web/gen-docs.sh`
+     - [XMonad mailing list](https://mail.haskell.org/mailman/listinfo/xmonad)
+     - [Haskell Cafe](https://mail.haskell.org/cgi-bin/mailman/listinfo/haskell-cafe)
```
Comment (slotThe) [inline MAINTAINERS.md:105]: Perhaps another cool place would be the Haskell Discourse

**Review by slotThe — APPROVED** [review]
This is awesome!  Thank you for improving the release process to this degree

File: MAINTAINERS.md (hunk 1 above)
Comment (liskin) [reply MAINTAINERS.md]: Yes, I will squash the fixup before merging :-)

File: MAINTAINERS.md (hunk 2 above)
Comment (liskin) [reply MAINTAINERS.md:105]: daf6d06

File: MAINTAINERS.md (hunk 1 above)
Comment (slotThe) [reply MAINTAINERS.md]: Oh!  Didn't realize there was a fixup already, sorry about that

**Comment by liskin** [comment]
I shall note that the release workflow here is somewhat different from the one that was used for the X11 release because there was an issue that needed to be fixed, and thus what is here is (to some extent) untested. We will have one opportunity to test it when releasing xmonad, and then another with xmonad-contrib. Then I'll make sure all workflows are the same.

I'll merge it now as there is nothing to be done before trying it on the actual release.

**Pull request merged**